### PR TITLE
{chem}[foss/2023a] Solids4foam v2.1

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/405

### CODE REVIEW

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/s/Solids4foam/Solids4foam-2.1-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+install_cmd = 'source $FOAM_BASH && '
+# variables of target install directory
+install_cmd += 'export WM_PROJECT_USER_DIR=%(installdir)s && '
+install_cmd += 'export FOAM_USER_APPBIN=%(installdir)s/bin && '
+install_cmd += 'export FOAM_USER_LIBBIN=%(installdir)s/lib && '
+install_cmd += 'export LD_LIBRARY_PATH=%(installdir)s/lib:$LD_LIBRARY_PATH && '
+install_cmd += 'export EIGEN_DIR=$EBROOTEIGEN && '
+install_cmd += 'export S4F_NO_FILE_FIXES=1 && ./Allwmake -j %(parallel)s'
+# copy tests
+install_cmd += ' && cp -av tutorials %(installdir)s'
```
Comment (boegel) [inline easybuild/easyconfigs/s/Solids4foam/Solids4foam-2.1-foss-2023a.eb:33]: @pavelToman This tells me that we could actually be using `MakeCp` here, and customize the build command via `build_cmd`:

```python
easyblock = 'MakeCp'

...
prebuildopts = "source $FOAM_BASH && "
...
build_cmd = './Allwmake'

files_to_copy = [...]
```

That way, the `-j %(parallel)s` is not required (will be added automatically I think), and the `extract_sources = True` can be dropped too.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/Solids4foam/Solids4foam-2.1-foss-2023a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/s/Solids4foam/Solids4foam-2.1-foss-2023a.eb:33]: I overlooked that the `Allwmake` already installs stuff in the installation directory that is specified via `$WM_PROJECT_USER_DIR` & co, so with that in mind `MakeCp` is even worse, so don't use that :)

Add a comment abovevthe `install_cmd` though to highlight that the `Allwmake` command doesn't only build stuff, but also installs stuff, and then we can proceed like this as is

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by laraPPr** [comment]
Test report by @laraPPr
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node4013.donphan.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.11.3
See https://gist.github.com/laraPPr/f0da2d54bda6998e6cf9ce75ef28884e for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3163.skitty.os - Linux RHEL 9.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.11.3
See https://gist.github.com/laraPPr/7de82e7afd3ff521ed5bc5419cf86125 for a full test report.

**Review by laraPPr — APPROVED** [review]
lgtm

**Comment by laraPPr** [comment]
Going in, thanks @pavelToman!

**Pull request merged**